pkg/networking/networking.go

```diff
@@ -126,6 +128,13 @@ func (n *networkImpl) AddHeaderField(key, value string) {
 }
 
 func (n *networkImpl) AddHeaders(request *http.Request) error {
+	// add snyk-request-id header with interaction id
+	uuid, err := GenerateInteractionUrn()
+	if err != nil {
+		n.logger.Warn().Err(err).Msg("Error generating Snyk request ID")
+		uuid = "urn:snyk:interaction:unknown"
+	}
+	n.AddHeaderField("snyk-request-id", uuid)
```

Comment:
Issues: 
1. This will generate a new request-id for each request, the goal is to have a common request id for each interaction, which can include multiple http requests.
2. snyk-request-id has to be a UUID not an URN, the idea is to use the same UUID from the request-id as for the InteractionID.
3. Having a constant request-id, will break any request tracing, in case of an error, it is better to not have a request-id at all.

Reply:
The integration should be on application level, where the interaction is defined. In the CLI this is when running a command, in and IDE, this might be the click on the scan button.

Reply:
Thank you! This cleared up things, I pushed the helper function.